azure-pipelines.yml

```diff
@@ -106,4 +224,68 @@ jobs:
       displayName: "Run tests"
       workingDirectory: $(Build.BinariesDirectory)/build
 
+
+  - job: WindowsBuck
+
+    pool:
+      vmImage: vs2017-win2016
+
+    steps:
+    - powershell: |
+        (New-Object System.Net.WebClient).DownloadFile("https://github.com/facebook/buck/releases/download/v2019.06.17.01/buck.2019.06.17.01.nupkg", `
+        "buck.2019.06.17.01.nupkg")
+        Install-Package buck -Source . -SkipDependencies
+      displayName: "Install buck"
+      workingDirectory: $(Build.BinariesDirectory)
+
+    - script: |
+        mklink /J $(Build.BinariesDirectory)\vsinstall "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise"
+
+    - powershell: |
+        $vcvers = cat $(Build.BinariesDirectory)\vsinstall\VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt
+        .\tools\generate_buck_config.ps1 -VsInstall '$(Build.BinariesDirectory)\vsinstall' `
+        -VcToolsVersion "$vcvers" `
+        -SdkInstall '' `
+        -SdkVersion '' `
+        -Python3Path 'C:\hostedtoolcache\windows\Python\3.7.2\x64\python.exe' `
+        -BuckConfigRoot .\tools\buckconfigs\
+      displayName: "Generate Buck config"
+
+    - script: |
+        C:\Python27amd64\python.exe "C:\Program Files\PackageManagement\NuGet\Packages\buck.2019.06.17.01\tools\buck.pex" build @mode/windows-x86_64/release osquery:osqueryd
+      displayName: "Build osquery"
+
+    - powershell: |
+        C:\Python27amd64\python.exe "C:\Program Files\PackageManagement\NuGet\Packages\buck.2019.06.17.01\tools\buck.pex" test @mode/windows-x86_64/release osquery/...
+      displayName: "Run tests"
+
+    - powershell: |
+        echo "##vso[task.setvariable variable=Status;isOutput=true]1"
+      name: JobResult
+
+
+  - job: Windows
+
+    pool:
+      vmImage: 'Ubuntu-16.04'
```

Comment:
Is this a placeholder?

Reply:
No, it's meant to "collect" the results of the Windows jobs.
So since status badges can only be created for the entire pipeline or a single job, when you have multiple jobs to check for status, in this case two, you need a third job with the name of the job you create the status badge from, which fails if one of the other two job fails, succeeds otherwise.
I didn't find a way to avoid using a VM unfortunately...

Reply:
So it's a `boolean and` to create a "windows" status badge? Do we need that? (vs having both status badges)

Reply:
The reason I'm doing that is simply space used on the README file to show the results.
This is also one of the reasons I unified the build and test jobs.
The table showing the statuses is already cramped with 3 statuses, with this PR we would have 6, and with the other PR for Debug builds it will become 8 initially, but when all the platforms and build systems will have then, it will be 12 status badges, which is quite a lot.

Reply:
Thanks for explaining!